src/program.cpp

```diff
@@ -426,6 +426,25 @@ void preview_argument(std::ostream& os, const argument& a)
         });
 }
 
+static bool is_compatible_shape(const shape& actual, const shape& expected)
```

Comment:
since this function is  only used inside `assert` it will not have any uses when doing Release builds. 
`assert` are only enabled for the Debug builds. 

You can do a fake call to `is_compatible_shape()` to avoid build error 
`(void)(is_compatible_shape(shape{}, shape{})`

Reply:
Is there any problem with wrapping it in a `#ifndef NDEBUG` instead?

Reply:
you can do that too. That's valid solution as well

Reply:
Marked this conversation unresolved again because I thought codecov would not flag debug-only code lines 446 and 451.  Why does it want this assertion code to be covered by a test?